diff --git a/node_modules/@rnmapbox/maps/android/build.gradle b/node_modules/@rnmapbox/maps/android/build.gradle
index 7de0f2f..b78b6b2 100644
--- a/node_modules/@rnmapbox/maps/android/build.gradle
+++ b/node_modules/@rnmapbox/maps/android/build.gradle
@@ -245,7 +245,7 @@ Please upgrade to Mapbox v11.x or use @rnmapbox/maps 10.2.x
     buildTypes {
         release {
             minifyEnabled false
-            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
         }
     }
 }
diff --git a/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/utils/DownloadMapImageTask.kt b/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/utils/DownloadMapImageTask.kt
index 2638b11..556a353 100644
--- a/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/utils/DownloadMapImageTask.kt
+++ b/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/utils/DownloadMapImageTask.kt
@@ -77,7 +77,7 @@ class DownloadMapImageTask(context: Context, map: MapboxMap, imageManager: Image
             result = DataSources.waitForFinalResult(dataSource)
             result?.get()?.let { image ->
                 if (image is CloseableStaticBitmap) {
-                    val bitmap = image.underlyingBitmap.copy(Bitmap.Config.ARGB_8888, true)
+                    val bitmap = image.underlyingBitmap?.copy(Bitmap.Config.ARGB_8888, true) ?: return@let null
                     bitmap.density = DisplayMetrics.DENSITY_DEFAULT
 
                     CoroutineScope(Dispatchers.Main).launch {
